source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,797 @@
+ *                   value).
+ * @param[in] duplex: The nature of the channel. This must be set to kPHY_FullDuplex by
+ *                   auto-negotiation.
+ *
+ * @return kStatus_Success if the ENET module was initialized; error code otherwise.
+ */
+static status_t xEMACInit( phy_speed_t speed,
+                           phy_duplex_t duplex )
+{
+    enet_config_t config;
+    uint32_t sysClock;
+    enet_buffer_config_t buffCfg[ ENET_RING_NUM ];
+    status_t xStatus;
+    uint32_t instance;
+    static ENET_Type * const enetBases[] = ENET_BASE_PTRS;
+    static const IRQn_Type enetTxIrqId[] = ENET_Transmit_IRQS;
+    /*! @brief Pointers to enet receive IRQ number for each instance. */
+    static const IRQn_Type enetRxIrqId[] = ENET_Receive_IRQS;
+    int i;
+
+    /* Make sure that the channel is full duplex. */
+    configASSERT( duplex == kPHY_FullDuplex );
+
+    /* prepare the buffer configuration. */
+    buffCfg[ 0 ].rxBdNumber = ENET_RXBD_NUM;                  /* Receive buffer descriptor number. */
+    buffCfg[ 0 ].txBdNumber = ENET_TXBD_NUM;                  /* Transmit buffer descriptor number. */
+    buffCfg[ 0 ].rxBuffSizeAlign = sizeof( rx_buffer_t );     /* Aligned receive data buffer size. */
+    buffCfg[ 0 ].txBuffSizeAlign = sizeof( tx_buffer_t );     /* Aligned transmit data buffer size. */
+    buffCfg[ 0 ].rxBdStartAddrAlign =
+        &( rxBuffDescrip_0[ 0 ] );                            /* Aligned receive buffer descriptor start address. */
+    buffCfg[ 0 ].txBdStartAddrAlign =
+        &( txBuffDescrip_0[ 0 ] );                            /* Aligned transmit buffer descriptor start address. */
+    buffCfg[ 0 ].rxBufferAlign =
+        &( rxDataBuff_0[ 0 ][ 0 ] );                          /* Receive data buffer start address. */
+    buffCfg[ 0 ].txBufferAlign = &( txDataBuff_0[ 0 ][ 0 ] ); /* Transmit data buffer start address. */
+    buffCfg[ 0 ].txFrameInfo = NULL;                          /* Transmit frame information start address. Set only if using zero-copy transmit. */
+    buffCfg[ 0 ].rxMaintainEnable = true;                     /* Receive buffer cache maintain. */
+    buffCfg[ 0 ].txMaintainEnable = true;                     /* Transmit buffer cache maintain. */
+
+    sysClock = phyHandle.mdioHandle->resource.csrClock_Hz;
+
+    ENET_GetDefaultConfig( &config );
+
+    config.ringNum = ENET_RING_NUM;
+    config.rxBuffAlloc = NULL;
+    config.rxBuffFree = NULL;
+    config.userData = ethernetifLocal;
+    config.miiSpeed = ( enet_mii_speed_t ) speed;
+    config.miiDuplex = ( enet_mii_duplex_t ) duplex;
+
+    /* Only get interrupt for incoming messages. */
+    config.interrupt = kENET_RxFrameInterrupt;
+    config.callback = ethernet_callback;
+
+    for( instance = 0; instance < ARRAY_SIZE( enetBases ); instance++ )
+    {
+        if( enetBases[ instance ] == ethernetifLocal->base )
+        {
+            NVIC_SetPriority( enetRxIrqId[ instance ], ENET_PRIORITY );
```

Comment:
Perhaps an NVIC_SetVector would be useful here?

Reply:
The vector handler is set inside the call to `ENET_Init`.